ReactAndroid/src/main/java/com/facebook/react/ReactRootView.java

```diff
@@ -468,6 +475,85 @@ public void setAppProperties(@Nullable Bundle appProperties) {
     }
   }
 
+  private static class RootViewInsets {
+    float top;
+    float right;
+    float bottom;
+    float left;
+
+    RootViewInsets(float top, float right, float bottom, float left) {
+      this.top = top;
+      this.right = right;
+      this.bottom = bottom;
+      this.left = left;
+    }
+  }
+
+  private RootViewInsets getSafeAreaInsets() {
+    // Window insets are parts of the window that are covered by system views (status bar,
+    // navigation bar, notches). There are no apis the get these values for android < M so we
+    // do a best effort polyfill.
+    // TODO: Use the DisplayCutout api when we target sdk 28+
+    RootViewInsets windowInsets;
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
```

Comment:
Why do you have this version check?`WindowInsets` is actually available from SDK level 20.